// @zetachain/toolkit/dist/packages/tasks/src/pools.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.poolsTask = void 0;
const utils_1 = require("ethers/lib/utils");
const config_1 = require("hardhat/config");
const src_1 = require("../../client/src/");
const main = async (args, hre) => {
    const client = new src_1.ZetaChainClient({
        network: args.mainnet ? "mainnet" : "testnet",
    });
    const pools = await client.getPools();
    const poolsDisplay = pools.map((pool) => {
        return {
            ...pool,
            t0: {
                ...pool.t0,
                reserve: (0, utils_1.formatUnits)(pool.t0.reserve, pool.t0.decimals),
            },
            t1: {
                ...pool.t1,
                reserve: (0, utils_1.formatUnits)(pool.t1.reserve, pool.t1.decimals),
            },
        };
    });
    const tableData = {};
    poolsDisplay.forEach((pool) => {
        const r0 = parseFloat(pool.t0.reserve);
        const r1 = parseFloat(pool.t1.reserve);
        tableData[pool.pair] = {
            Pool: `${pool.t0.symbol} / ${pool.t1.symbol}`,
            Reserves: `${r0} / ${r1}`,
        };
    });
    if (args.json) {
        console.log(pools);
    }
    else {
        console.table(tableData);
    }
};
exports.poolsTask = (0, config_1.task)("pools", "", main)
    .addFlag("json", "Print the result in JSON format")
    .addFlag("mainnet", "Run the task on mainnet");
