// @zetachain/toolkit/dist/typechain-types/factories/@zetachain/protocol-contracts/contracts/zevm/interfaces/ZContract__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZContract__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "bytes",
                        name: "origin",
                        type: "bytes",
                    },
                    {
                        internalType: "address",
                        name: "sender",
                        type: "address",
                    },
                    {
                        internalType: "uint256",
                        name: "chainID",
                        type: "uint256",
                    },
                ],
                internalType: "struct zContext",
                name: "context",
                type: "tuple",
            },
            {
                internalType: "address",
                name: "zrc20",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "amount",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "message",
                type: "bytes",
            },
        ],
        name: "onCrossChainCall",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class ZContract__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ZContract__factory = ZContract__factory;
ZContract__factory.abi = _abi;
