// @zetachain/toolkit/dist/typechain-types/factories/contracts/shared/libraries/UniswapV2Library__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniswapV2Library__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "IdenticalAddresses",
        type: "error",
    },
    {
        inputs: [],
        name: "InsufficientAmount",
        type: "error",
    },
    {
        inputs: [],
        name: "InsufficientInputAmount",
        type: "error",
    },
    {
        inputs: [],
        name: "InsufficientLiquidity",
        type: "error",
    },
    {
        inputs: [],
        name: "InsufficientOutputAmount",
        type: "error",
    },
    {
        inputs: [],
        name: "InvalidPath",
        type: "error",
    },
    {
        inputs: [],
        name: "ZeroAddress",
        type: "error",
    },
];
const _bytecode = "0x60566050600b82828239805160001a6073146043577f4e487b7100000000000000000000000000000000000000000000000000000000600052600060045260246000fd5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea26469706673582212201b89ee00e541705543f8a1c99c22642eb1f3b4269216412ce03159e60997c13f64736f6c63430008070033";
const isSuperArgs = (xs) => xs.length > 1;
class UniswapV2Library__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.UniswapV2Library__factory = UniswapV2Library__factory;
UniswapV2Library__factory.bytecode = _bytecode;
UniswapV2Library__factory.abi = _abi;
