// @zetachain/toolkit/dist/typechain-types/factories/@zetachain/protocol-contracts/contracts/zevm/SystemContract.sol/SystemContractErrors__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemContractErrors__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "CallerIsNotFungibleModule",
        type: "error",
    },
    {
        inputs: [],
        name: "CantBeIdenticalAddresses",
        type: "error",
    },
    {
        inputs: [],
        name: "CantBeZeroAddress",
        type: "error",
    },
    {
        inputs: [],
        name: "InvalidTarget",
        type: "error",
    },
    {
        inputs: [],
        name: "ZeroAddress",
        type: "error",
    },
];
class SystemContractErrors__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.SystemContractErrors__factory = SystemContractErrors__factory;
SystemContractErrors__factory.abi = _abi;
