// @zetachain/toolkit/dist/typechain-types/factories/contracts/shared/libraries/SafeMath.sol/Math__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Math__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "AdditionsOverflow",
        type: "error",
    },
    {
        inputs: [],
        name: "MultiplicationsOverflow",
        type: "error",
    },
    {
        inputs: [],
        name: "SubtractionsUnderflow",
        type: "error",
    },
];
const _bytecode = "0x60566050600b82828239805160001a6073146043577f4e487b7100000000000000000000000000000000000000000000000000000000600052600060045260246000fd5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea264697066735822122004e4099cea2415c66ea8c077ff18561866ce887a8464ad7469d8dfd337fc637c64736f6c63430008070033";
const isSuperArgs = (xs) => xs.length > 1;
class Math__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.Math__factory = Math__factory;
Math__factory.bytecode = _bytecode;
Math__factory.abi = _abi;
