// @zetachain/toolkit/dist/typechain-types/factories/contracts/shared/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WZETA__factory = exports.TestUniswapRouter__factory = exports.MockZRC20__factory = exports.libraries = exports.interfaces = exports.mockSystemContractSol = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.mockSystemContractSol = __importStar(require("./MockSystemContract.sol"));
exports.interfaces = __importStar(require("./interfaces"));
exports.libraries = __importStar(require("./libraries"));
var MockZRC20__factory_1 = require("./MockZRC20__factory");
Object.defineProperty(exports, "MockZRC20__factory", { enumerable: true, get: function () { return MockZRC20__factory_1.MockZRC20__factory; } });
var TestUniswapRouter__factory_1 = require("./TestUniswapRouter__factory");
Object.defineProperty(exports, "TestUniswapRouter__factory", { enumerable: true, get: function () { return TestUniswapRouter__factory_1.TestUniswapRouter__factory; } });
var WZETA__factory_1 = require("./WZETA__factory");
Object.defineProperty(exports, "WZETA__factory", { enumerable: true, get: function () { return WZETA__factory_1.WZETA__factory; } });
