// @zetachain/toolkit/dist/typechain-types/factories/@zetachain/protocol-contracts/contracts/evm/interfaces/ZetaInterfaces.sol/ZetaConnector__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZetaConnector__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "uint256",
                        name: "destinationChainId",
                        type: "uint256",
                    },
                    {
                        internalType: "bytes",
                        name: "destinationAddress",
                        type: "bytes",
                    },
                    {
                        internalType: "uint256",
                        name: "destinationGasLimit",
                        type: "uint256",
                    },
                    {
                        internalType: "bytes",
                        name: "message",
                        type: "bytes",
                    },
                    {
                        internalType: "uint256",
                        name: "zetaValueAndGas",
                        type: "uint256",
                    },
                    {
                        internalType: "bytes",
                        name: "zetaParams",
                        type: "bytes",
                    },
                ],
                internalType: "struct ZetaInterfaces.SendInput",
                name: "input",
                type: "tuple",
            },
        ],
        name: "send",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class ZetaConnector__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ZetaConnector__factory = ZetaConnector__factory;
ZetaConnector__factory.abi = _abi;
