// @zetachain/toolkit/dist/typechain-types/factories/@zetachain/protocol-contracts/contracts/evm/interfaces/ZetaInterfaces.sol/ZetaReceiver__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZetaReceiver__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "bytes",
                        name: "zetaTxSenderAddress",
                        type: "bytes",
                    },
                    {
                        internalType: "uint256",
                        name: "sourceChainId",
                        type: "uint256",
                    },
                    {
                        internalType: "address",
                        name: "destinationAddress",
                        type: "address",
                    },
                    {
                        internalType: "uint256",
                        name: "zetaValue",
                        type: "uint256",
                    },
                    {
                        internalType: "bytes",
                        name: "message",
                        type: "bytes",
                    },
                ],
                internalType: "struct ZetaInterfaces.ZetaMessage",
                name: "zetaMessage",
                type: "tuple",
            },
        ],
        name: "onZetaMessage",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "address",
                        name: "zetaTxSenderAddress",
                        type: "address",
                    },
                    {
                        internalType: "uint256",
                        name: "sourceChainId",
                        type: "uint256",
                    },
                    {
                        internalType: "bytes",
                        name: "destinationAddress",
                        type: "bytes",
                    },
                    {
                        internalType: "uint256",
                        name: "destinationChainId",
                        type: "uint256",
                    },
                    {
                        internalType: "uint256",
                        name: "remainingZetaValue",
                        type: "uint256",
                    },
                    {
                        internalType: "bytes",
                        name: "message",
                        type: "bytes",
                    },
                ],
                internalType: "struct ZetaInterfaces.ZetaRevert",
                name: "zetaRevert",
                type: "tuple",
            },
        ],
        name: "onZetaRevert",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class ZetaReceiver__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ZetaReceiver__factory = ZetaReceiver__factory;
ZetaReceiver__factory.abi = _abi;
