// @zetachain/toolkit/dist/typechain-types/factories/contracts/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestZRC20__factory = exports.TestSystemContract__factory = exports.SwapHelperLib__factory = exports.OnlySystem__factory = exports.shared = exports.zetaConnectorMockSol = exports.ethZetaMockSol = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.ethZetaMockSol = __importStar(require("./EthZetaMock.sol"));
exports.zetaConnectorMockSol = __importStar(require("./ZetaConnectorMock.sol"));
exports.shared = __importStar(require("./shared"));
var OnlySystem__factory_1 = require("./OnlySystem__factory");
Object.defineProperty(exports, "OnlySystem__factory", { enumerable: true, get: function () { return OnlySystem__factory_1.OnlySystem__factory; } });
var SwapHelperLib__factory_1 = require("./SwapHelperLib__factory");
Object.defineProperty(exports, "SwapHelperLib__factory", { enumerable: true, get: function () { return SwapHelperLib__factory_1.SwapHelperLib__factory; } });
var TestSystemContract__factory_1 = require("./TestSystemContract__factory");
Object.defineProperty(exports, "TestSystemContract__factory", { enumerable: true, get: function () { return TestSystemContract__factory_1.TestSystemContract__factory; } });
var TestZRC20__factory_1 = require("./TestZRC20__factory");
Object.defineProperty(exports, "TestZRC20__factory", { enumerable: true, get: function () { return TestZRC20__factory_1.TestZRC20__factory; } });
